### DISCUSSION TITLE
Several wrong routes chosen compared to OsmAnd. Why?

### DISCUSSION THREAD

**Discussion by ghost** [organicmaps/organicmaps#5158] [Q&A]
Nevermind, apparently not worth the trouble to report these. Better stick with OsmAnd.

**Comment by biodranik** [comment]
Thanks! Can you please provide HTML links to the same routes in OpenStreetMap.org for easier debugging?

@vng PTAL

**Comment by ghost** [comment]


**Reply by biodranik** [reply]
It also makes sense to check underlying OSM maps data. But I assume that the issue is caused by missing penalization of turns (and traffic lights) in Organic Maps. We'll implement it later.

**Reply by biodranik** [reply]
Another reason could be that we're ignoring elevation for cars at the moment. @vng is it correct?

**Comment by vng** [comment]
1. Worth adding smoothness=very_bad (or horrible) to the small residential road. OSRM and GraphHopper also use this route.
2. OM bug, should check tertiary vs secondary profile. The secondary road should be better here.
3. If the road is completely gone, it should be deleted from OSM (or tagged respectively). Now there is no difference (except + 1 turn) between those 2 routes. Both use tertiary roads, both are the same length.

**Comment by vng** [comment]
@jammynads Please, fill free to add here other examples (from your home town) where OM makes bad routes.

**Comment by vng** [comment]
Oh, looks like all examples are gone .. Try to restore:
- Car route (35.5170594, 24.0938699) -> (35.5446109, 24.1312439) should use secondary road (EO94) instead of tertiary
- Bicycle route (61.5445696, 23.9394003) -> (61.6153965, 23.876755) should use "Eerolansuorantie" tertiary road instead of secondary "338".
- Bicycle route (61.49610, 23.99260) -> (61.51920, 24.01730) has short option using highway=track (+mud) or long option using cycleway.
https://www.openstreetmap.org/directions?engine=graphhopper_bicycle&route=61.4961%2C23.9926%3B61.5192%2C24.0173#map=13/61.4992/24.0242&layers=D
https://www.openstreetmap.org/directions?engine=fossgis_valhalla_bicycle&route=61.4961%2C23.9926%3B61.5192%2C24.0173#map=14/61.5075/24.0060&layers=D
- Bicycle route (61.4712942, 23.9417559) -> (61.4580572, 23.9129292) makes big detour instead of using short but unpaved (actually compacted) residential.
https://www.openstreetmap.org/directions?engine=graphhopper_bicycle&route=61.4703%2C23.9417%3B61.4579%2C23.9132#map=15/61.4641/23.9268
https://www.openstreetmap.org/directions?engine=fossgis_valhalla_bicycle&route=61.4703%2C23.9417%3B61.4579%2C23.9132#map=13/61.4679/23.9056

**Reply by pastk** [reply]
>     * Bicycle route (61.5445696, 23.9394003) -> (61.6153965, 23.876755) should use "Eerolansuorantie" tertiary road instead of secondary "338".
> 

It prefers tertiary now because of https://github.com/organicmaps/organicmaps/pull/5192#discussion_r1826000716 i.e. the secondary gets a penalty because it has explicit `cycleway:both=no`. But the tertiary doesn't have a cycleway also its just not explicit!

Its still makes sense to prefer a tertiary just because its generally calmer and lower traffic, we can do it by adjusting weights a little (now they're same for secondaries and tertiaries). Also it would make sense to consider maxspeed (in this case the tertiary has 50-60 speed limit, while secondary has 80).

>     * Bicycle route (61.49610, 23.99260) -> (61.51920, 24.01730) has short option using highway=track (+mud) or long option using cycleway.

Details:
a long option is 12km 40min, a short option is 4km 27min

>     * Bicycle route (61.4712942, 23.9417559) -> (61.4580572, 23.9129292) makes big detour instead of using short but unpaved (actually compacted) residential.

This one works as expected now.
